### PR TITLE
Ingress Api Change in Pod  

### PR DESCRIPTION
**Pull request by gatici**
Signed-off-by: aticig <gulsum.atici@canonical.com>

### CODE REVIEW

**Comment by gatici** [comment]
This change fixes the error "k8s pod-spec-set charm erroring with "server could not find the requested resource" in microk8s version 1.22 and higher versions. We are changing ingress resource structure to become compatible with api version v1 according to "https://bugs.launchpad.net/juju/+bug/1939278"

File: opslib/osm/pod.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def ingress_resource(self):
             r["spec"]["tls"] = self.tls
         return r
 
-    def add_rule(self, hostname: str, service_name, port, path: str = "/"):
+    def add_rule(self, hostname: str, service_name, path_type, port, path: str = "/",):
```
Comment (davigar15) [inline opslib/osm/pod.py]: give it a default value of "Prefix". This way we don't need to change every charm.

**Review by davigar15 — CHANGES_REQUESTED** [review]
Please, comment if you have confirmed the patch works in K8s 1.23

File: opslib/osm/pod.py (hunk 1 above)
Comment (davigar15) [reply opslib/osm/pod.py]: also, execute `black opslib/`.

**Comment by gatici** [comment]
This is not valid anymore.

**Pull request closed without merging**